### PR TITLE
feat(qwen-vl): add Qwen3.6 verification card

### PR DESCRIPTION
**Pull request by cuichenx**
## Summary

- add a complete verification card for the canonical multimodal checkpoint
  `Qwen/Qwen3.6-35B-A3B` at immutable revision
  `995ad96eacd98c81ed38be0c5b274b04031597b0`
- document that Qwen3.5 and Qwen3.6 use the same existing
  `qwen3_5_moe` vision-language architecture; no Qwen3.6-specific recipe is
  added
- verify conversion, multimodal inference, MedPix SFT and PEFT, post-SFT
  export/inference, long-context packed SFT, bounded DataComp projection
  pretraining, full-state checkpoint resume, and a canonical GB200 MXFP8
  performance workload
- keep public workflows on maintained conversion, training, and inference
  launchers, with resource-proportional Slurm requests and no implicit
  exclusivity

## Identity and model evidence

The public model name has no `VL` suffix, but Hugging Face classifies it as
image-text-to-text and exposes `Qwen3_5MoeForConditionalGeneration`. The
existing Bridge intentionally serves Qwen3.5 and Qwen3.6 VLM checkpoints while
deriving each checkpoint's dimensions and MoE layout from its config.

Verified model-level items include:

- CPU and distributed-GPU HF-to-Megatron import and Megatron-to-HF export
- persisted CPU bitwise round-trip of all 1,045 BF16 tensors and
  71,903,645,408 tensor bytes
- distributed-GPU round-trip of all 1,045 tensors at `atol=0.1`, `rtol=1e-5`
- strict Transformers reload of 26 indexed BF16 shards
- HF/Megatron multimodal forward comparison with a matching next token and
  cosine similarity 0.999156
- deterministic processor-native MedPix Megatron inference

## Training and performance evidence

Every H100 training inventory leaf in the card is verified:

- 100-step MedPix SFT, full checkpoints at steps 50 and 100, and successful
  fresh-process export
- 100-step MedPix LoRA, adapter checkpoint save, and successful fresh-process
  reload
- post-SFT HF export, strict reload, and deterministic multimodal inference
- 20-step MedPix long-context SFT using 8192-token deferred in-batch packing
  and CP2
- deterministic 20-step DataComp reference plus direct step-10 restart with
  optimizer, scheduler, RNG, and data-order state restoration

The DataComp leaf is deliberately scoped to frozen-tower projection
pretraining support, not canonical DataComp/CLIP convergence. Its reusable
Energon preparation support is tracked separately in #5087.

The final matched 8x GB200 MedPix SFT and PEFT runs came from clean commit
`d9b4f9168b247b1068897358f061d82e63f8f1ef`. Their persisted configs differ
only in the evidence-output path, intentional LR/min-LR, and LoRA block. The
step-1 losses were 2.212866 and 2.214827, an absolute difference of 0.001961
(0.088618%), so the higher final PEFT loss is not an initial model/data/
topology/schedule mismatch.

The canonical 8x GB200 MXFP8 mock-data performance recipe completed 50 finite
steps with zero skipped or NaN iterations. The persisted post-setup config
confirmed MBS3/GBS480, HybridEP, forced expert balancing, cuDNN LayerNorm, and
Transformer Engine MoE router/preprocessing graph scopes. Steps 41-50 averaged
26,880.420 ms and 210.360 model TFLOP/s/GPU. This is synthetic throughput
evidence, not convergence evidence.

## Recipe and resource safety

- required functional defaults live in the shared Qwen3.5/Qwen3.6 recipes;
  card commands retain only workload identity, bounded schedule/data,
  artifact, and evidence-output values
- BF16 and FP8 current-scaling recipes remain eager; only the measured MXFP8
  specialization enables scoped Transformer Engine graphs
- card commands use maintained `train.sh`, `convert.sh`, and inference
  launchers rather than custom model-specific Slurm scripts
- no card command requests `--exclusive`; CPU-focused validation requests no
  GPU, and training requests only the GPUs they use
- public card content contains no private runtime paths, accounts,
  credentials, or Slurm job metadata

## Validation

- model-verification-card validator: passed
- focused Qwen VLM recipe, conversion launcher, and performance-recipe suite:
  573 passed
- `git diff --check`: passed
- full pre-commit hook set in `nvcr.io/nvidia/nemo:26.06`: passed
- independent final-diff review: no actionable findings
- DCO sign-off: present

The mandatory host command `uv run pre-commit run --all-files` is blocked
before hook startup because locked `nvidia-resiliency-ext==0.6.0` publishes a
manylinux 2.39 wheel while the host is manylinux 2.31. The complete hook set
passes in the project-compatible NeMo 26.06 container.

### CODE REVIEW

File: scripts/conversion/arguments.py [hunk 1]
```diff
@@ -41,13 +41,18 @@ def _add_execution_arguments(parser: argparse.ArgumentParser, *, default_device:
...
-        help="GPUs per node; required for the GPU backend.",
+        help="GPUs per node; required for the GPU backend and optional as a CPU-backend runtime resource.",
     )
     execution.add_argument("--mem", default="0", help="Slurm memory request (default: 0, all node memory).")
     execution.add_argument("--account", default=os.environ.get("SLURM_ACCOUNT"), help="Slurm account.")
     execution.add_argument("--partition", default=os.environ.get("SLURM_PARTITION"), help="Slurm partition.")
     execution.add_argument("--time", default="04:00:00", help="Slurm time limit (default: 04:00:00).")
     execution.add_argument("--gres", help="Optional Slurm GRES value for GPU jobs.")
+    execution.add_argument(
+        "--exclusive",
```
Comment (cuichenx) [inline scripts/conversion/arguments.py:52]: exclusive flag is not used anywhere?

File: scripts/conversion/setup_conversion.py [hunk 2]
```diff
@@ -167,7 +172,7 @@ def _build_executor(
         nodes=args.nodes,
         ntasks_per_node=task_count,
         mem=args.mem,
-        exclusive=True,
+        exclusive=True if args.exclusive else None,
```
Comment (cuichenx) [inline scripts/conversion/setup_conversion.py]: why not exclusive=args.exclusive?

File: scripts/conversion/arguments.py (hunk 1 above)
Comment (cuichenx) [reply scripts/conversion/arguments.py:52]: Addressed in fe13c3d9d: `_build_executor` now passes `exclusive=args.exclusive`, so the flag is consumed directly and the default remains a shared allocation.

File: scripts/conversion/setup_conversion.py (hunk 2 above)
Comment (cuichenx) [reply scripts/conversion/setup_conversion.py]: Updated to `exclusive=args.exclusive` in fe13c3d9d, with focused assertions covering the default shared and explicit exclusive cases.

File: examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        dataset.pad_to_max_length=true dataset.enable_in_batch_packing=false
+        dataset.defer_in_batch_packing_to_step=true
+        scheduler.lr_decay_iters=100 dist.distributed_timeout_minutes=30
+        model.hf_model_id=Qwen/Qwen3.6-35B-A3B
+        model.bos_token_id=248044 model.eos_token_id=248044
+        model.moe_router_force_load_balancing=false
+        checkpoint.hf_source_path=Qwen/Qwen3.6-35B-A3B checkpoint.load=null
+        validation.eval_iters=0 validation.eval_interval=0
+        ddp.check_for_nan_in_grad=true ddp.check_for_large_grads=true
+        rerun_state_machine.check_for_nan_in_loss=true
```
Comment (cuichenx) [inline examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml]: don't override, make some of these default arguments in the recipe

File: scripts/training/setup_experiment.py [hunk 4]
```diff
@@ -74,7 +74,17 @@ def _build_parser() -> argparse.ArgumentParser:
...
     execution.add_argument("--partition", default=os.environ.get("SLURM_PARTITION"), help="Slurm partition.")
     execution.add_argument("--time", default="04:00:00", help="Slurm time limit.")
+    execution.add_argument(
+        "--mem",
+        default="0",
+        help="Slurm memory request per node, for example 512G. Defaults to all node memory.",
+    )
     execution.add_argument("--gres", help="Optional Slurm GRES value.")
+    execution.add_argument(
+        "--exclusive",
```
Comment (cuichenx) [inline scripts/training/setup_experiment.py]: add both --mem and --exclusive to a "additional_parameters" field.

File: src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/transformer_config.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+# Copyright (c) 2025-2026, NVIDIA CORPORATION.  All rights reserved.
```
Comment (cuichenx) [inline src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/transformer_config.py]: don't change copyright

File: scripts/training/setup_experiment.py (hunk 4 above)
Comment (cuichenx) [reply scripts/training/setup_experiment.py]: Addressed in 13b94ab7f. Both mem and explicit-only exclusive now live in executor.additional_parameters. The default has no exclusive entry/flag; --exclusive adds boolean true, which NeMo Run renders as a valueless #SBATCH --exclusive directive. Focused launcher tests cover default, partial-memory, and explicit-exclusive cases.

File: examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml (hunk 3 above)
Comment (cuichenx) [reply examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml]: Addressed in 13b94ab7f. I removed recipe-owned/historical-default values from both SFT and PEFT card commands, retained the run-specific Qwen3.6/MedPix/schedule/validation/logging/safety overrides, and added focused recipe assertions (plus explicit NaN-loss ownership) for the defaults the card now relies on.

File: src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/transformer_config.py (hunk 5 above)
Comment (cuichenx) [reply src/megatron/bridge/models/qwen_vl/modelling_qwen3_vl/transformer_config.py]: Addressed in 13b94ab7f. Restored the original 2025 copyright header. The separate Qwen3.5/Qwen3.6 shared-architecture documentation remains unchanged.

File: examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+  vision-language Bridge implementation. The common architecture is configured
+  dynamically from each checkpoint; the existing Qwen3.5 35B recipe remains a
+  Qwen3.5 checkpoint preset and is not relabeled as Qwen3.6. Functional
+  finetuning uses mmoukouba/MedPix-VQA at immutable revision
+  69eb219d477ab8699296da1fd6b89beb768042d3 through processor-native image
+  and text conversations. CPU and GPU import/export, round-trip weight
+  verification, forward-pass correlation, deterministic multimodal inference,
+  bounded 100-step MedPix SFT and PEFT runs with reloadable checkpoints, and
+  post-SFT HF export/inference were run; other items retain their explicit
+  status below.
```
Comment (cuichenx) [inline examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml]: summary should be shorter. don't regurgitate everything

Comment (cuichenx) [reply examples/model_verification_cards/qwen3.6-35b-a3b/card.yaml]: Addressed in fecd70c30. The summary is now only the required performance disclaimer plus one concise sentence covering verified and unverified scope; the detailed identity, evidence, and metrics remain in their item records.

**Comment by cuichenx** [comment]
Addressed the two remaining review threads in `8e9c51ad9`:

- **Card command length:** audited all commands, removed redundant recipe defaults, and moved the bounded GB200 SFT/PEFT topology, batch, optimizer, safety, and execution defaults into the shared Qwen3.5/Qwen3.6 library recipes. Card commands retain only model/data identity, bounded schedule, artifact, and persisted-config values.
- **PEFT versus SFT loss:** reran an exact clean matched 8x GB200 pair. Persisted configs differ only in evidence-output path, intentional LR/min-LR, and LoRA. Step-1 losses were 2.212866 versus 2.214827, an absolute difference of 0.001961 (0.088618%), ruling out an initial model/data/topology/schedule mismatch. The later loss difference is consistent with the different trainable parameter set and learning rate.

The final card validator, `git diff --check`, all-files pre-commit, focused 573-test suite, and independent final-diff review are clean.

**Comment by cuichenx** [comment]
/ok to test https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/5055/commits/8e9c51ad9e43e0a5a3e281df0eff917ef3d1d044

File: scripts/conversion/setup_conversion.py [hunk 7]
```diff
@@ -74,15 +74,18 @@ def _validate_args(args: argparse.Namespace) -> None:
         raise ValueError("--distributed-timeout-minutes must be at least 1.")
     if any(not value.strip() for value in args.srun_args):
         raise ValueError("--srun-arg values must not be empty.")
-    for name in ("tp", "pp", "ep", "etp"):
-        if getattr(args, name) < 1:
-            raise ValueError(f"--{name} must be at least 1.")
+    if args.command != "compare-hf":
```
Comment (yaoyu-33) [inline scripts/conversion/setup_conversion.py]: remove no need here
